R/set_args.R

```diff
@@ -45,4 +45,112 @@ set_inherited_args <- function(fun.name, inherited.args, ..., package='gsplot'){
   # // shed non-formals
   inherited.args = function_args(package, fun.name, inherited.args, drop=TRUE)
   return(c(inherited.args, set_args(fun.name, ..., package = package)))
-}
\ No newline at end of file
+}
+
+set_legend_args <- function(object, fun.name, ...) {
+  paramsAll <- c(...)
+  
+  if(any(names(paramsAll) %in% "legend.name")) {
+  
+    names(paramsAll)[which(names(paramsAll)=='legend.name')] <- 'legend'
+    default.args <- formals(graphics::legend)
+    overall.legend <- c("x", "y", "bty", "bg", "box.lty", "box.lwd", "box.col", "cex","xjust", 
+                        "yjust", "adj", "text.width", "merge", "trace", "plot", "ncol", 
+                        "horiz", "title", "inset", "title.col", "title.adj", "xpd")  
+    not.overall <- default.args[which(!names(default.args) %in% overall.legend)]
```

Comment:
make sure to create a test for this - static lists don't reflect any updates that base R might have.